java/ql/src/semmle/code/java/frameworks/spring/SpringWebUtil.qll

```diff
@@ -0,0 +1,241 @@
+        "org.springframework.web.util;ContentCachingRequestWrapper;false;ContentCachingRequestWrapper;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;ContentCachingRequestWrapper;false;getContentAsByteArray;;;Argument[-1];ReturnValue;taint",
+        // note for review: the `content` of this wrapper is determined by what's written to it; there's a function `getWriter()`
+        //   that gets the writer used to write to it; should I link these up outside of the csv models?
+        "org.springframework.web.util;ContentCachingResponseWrapper;false;ContentCachingResponseWrapper;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;ContentCachingResponseWrapper;false;getContentAsByteArray;;;Argument[-1];ReturnValue;taint",
+        "org.springframework.web.util;ContentCachingResponseWrapper;false;getContentInputStream;;;Argument[-1];ReturnValue;taint",
+        // note for review: adds a cookie (argument 1) to an `HttpServletResponse` (argument 0); whether a cookie should taint a response
+        //   is somewhat unclear to me.
+        "org.springframework.web.util;CookieGenerator;false;addCookie;;;Argument[-1];Argument[0];taint",
+        "org.springframework.web.util;CookieGenerator;false;addCookie;;;Argument[1];Argument[0];taint",
+        "org.springframework.web.util;CookieGenerator;false;getCookieDomain;;;Argument[-1];ReturnValue;taint",
+        "org.springframework.web.util;CookieGenerator;false;setCookieDomain;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;CookieGenerator;false;getCookieName;;;Argument[-1];ReturnValue;taint",
+        "org.springframework.web.util;CookieGenerator;false;setCookieName;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;CookieGenerator;false;getCookiePath;;;Argument[-1];ReturnValue;taint",
+        "org.springframework.web.util;CookieGenerator;false;setCookiePath;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;DefaultUriBuilderFactory;false;DefaultUriBuilderFactory;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;DefaultUriBuilderFactory;false;builder;;;Argument[-1];ReturnValue;taint",
+        "org.springframework.web.util;DefaultUriBuilderFactory;false;expand;;;Argument[-1..0];ReturnValue;taint",
+        "org.springframework.web.util;DefaultUriBuilderFactory;false;expand;(String,Map);;MapValue of Argument[1];ReturnValue;taint",
+        "org.springframework.web.util;DefaultUriBuilderFactory;false;expand;(String,Object[]);;ArrayElement of Argument[1];ReturnValue;taint",
+        "org.springframework.web.util;DefaultUriBuilderFactory;false;getDefaultUriVariables;;;Argument[-1];MapValue of ReturnValue;taint",
+        "org.springframework.web.util;DefaultUriBuilderFactory;false;setDefaultUriVariables;;;MapValue of Argument[0];Argument[-1];ReturnValue;taint",
```

Comment:
Here's the offending row.
`...MapValue of Argument[0];Argument[-1];ReturnValue...`